pkg/handlers/handlers.go

```diff
@@ -64,36 +64,67 @@ func newSizedMimetypeReaderFromFileReader(r fileReader) (sizedMimeTypeReader, er
 
 	size, err := r.Size()
 	if err != nil {
-		return sizedMimeTypeReader{}, fmt.Errorf("error getting file size: %w", err)
+		return sizedReader{}, fmt.Errorf("error getting file size: %w", err)
 	}
 
-	return sizedMimeTypeReader{
-		mimeExt:  r.mime.Extension(),
-		mimeType: mimeType(r.mime.String()),
-		size:     size,
-		Reader:   r.BufferedReadSeeker,
-	}, nil
+	return sizedReader{size: size, Reader: r.BufferedReadSeeker}, nil
 }
 
-// newSizedMimeTypeReader creates a new sizedMimeTypeReader from an io.Reader.
-// It uses a bufio.Reader to perform MIME type detection on the input reader
-// without consuming it, by peeking into the first 512 bytes of the input.
-// This encapsulates both the original reader and the detected MIME type information.
+// newSizedReader creates a new sizedReader from an io.Reader.
 // This function is particularly useful for specialized archive handlers
 // that need to pass extracted content to the default handler without modifying the original reader.
-func newSizedMimeTypeReader(r io.Reader, size int64) (sizedMimeTypeReader, error) {
+func newSizedReader(r io.Reader, size int64) (sizedReader, error) {
```

Comment:
It would never occur to me that this function could fail because of an unsupported file type - that behavior isn't hinted at in the name or the doc comment.